### PR TITLE
Decouple cucumber report footer from landing page and individual modules

### PR DESCRIPTION
**Pull request by bpvstaty366**

### CODE REVIEW

File: cypress/plugins/index.js [hunk 1]
```diff
@@ -298,7 +298,10 @@ module.exports = async (on, config) => {
             jsonReport = readDataFromFile(filePath + fileName);
           }
           const reportProperties = {};
+          const customFooter = require("../fixtures/customFooter.json");
```
Comment (parag-pv) [inline cypress/plugins/index.js:301]: Can use cy.fixture.then() instead of require

**Comment by parag-pv** [comment]
@bpvstaty366 Question: Is this only for combined runs or does it work for individual module runs? Say if only account is being run, will the inner page of account feature have the custom footer or the common footer from the home page?

**Comment by parag-pv** [comment]
Question 2: The common footer in the landing page is right now fetched from the reportEnv.json in Lambda, is that something that can be overriden in FCS? We need the ability to customize the common footer as well without actually modifying anything in Lambda

File: cypress/plugins/index.js (hunk 1 above)
Comment (bpvstaty366) [reply cypress/plugins/index.js:301]: no, can't use cy in after:run because of ReferenceError: cy is not defined

**Comment by parag-pv** [comment]
Can we put the `customReportFooters.json` in configModule's `cypress/fixtures`?

- Keep a base file in FCS but override it with whatever value is provided in configModule. 
- If a configModule does not have `customReportFooters.json`, use the data present in FCS.
- Override the entire file for now, per key based override should be implemented in [FIRECERT-2095](https://ccp.sys.comcast.net/browse/FIRECERT-2095), after which, the logic to override the footer data added in this ticket should be removed and the common logic from 2095 should be used.

File: cypress/fixtures/customReportData.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "customFooter": {
+    "PageFooter": "<div id=\"pagefooter\"><p><b>The following module exclusions apply:</b><ul><li>Discovery.launch (Due to launch crashes in Ripple 1 gateway with launcher builds)</li><li>Federated Data (Not currently supported as a rippleRPC only method)</li><li>Secure Storage( Test cases which involves passing empty string for key comes under distributorCertification and not under fireboltCertification)</li><li>Any test case which requires thunder events (denoted by @thunderproxy tag)</li><li>Any test case which requires a test step that is not possible with the methods we have available (denoted by @notSupported tag)</li><li>Lifecycle Visibility state validation fix :  Add the expected value in configModule file ( cypress.config.js) in the format &quot;visibilityState&quot; : { &quot;state&quot; : &quot;expectedValue&quot;}</li></ul></p></div>",
```
[the remark this answers is not in the document]
Comment (bpvstaty366) [reply cypress/fixtures/customReportData.json]: updated

File: cypress/fixtures/customReportData.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+  "customFooter": {
+    "PageFooter": "Custom footer for main page",
+    "Account": "<div><p><b>Custom footer for Account pages</b></p></div>",
+    "Device": "<div><p><b>Custom footer for Device pages</b></p></div>"
+  },
+  "customMetadata": {
+    "Account": {
+      "Setup": {
+        "PreReq1": "SetupAccount1",
```
Comment (parag-pv) [inline cypress/fixtures/customReportData.json:10]: We might not need PreReq 1, 2 and 3. A single key under moduleName would've sufficed, if the value can contain line breaks.
Example:
`
"customMetadata": {
    "Account": "SetUpAccount1</br>SetupAccount2......",
    "Device": "SetupDevice1</br>SetupDevice2...."
}
`
These do not have to be <li> list items, a <p> paragraph would suffice as mentioned in ticket description.

File: cypress/plugins/index.js [hunk 4]
```diff
@@ -298,7 +298,14 @@ module.exports = async (on, config) => {
             jsonReport = readDataFromFile(filePath + fileName);
           }
           const reportProperties = {};
+          let customReportData;
+          try {
+            customReportData = require('../../node_modules/configModule/cypress/fixtures/objects/customReportData.json');
```
Comment (parag-pv) [inline cypress/plugins/index.js:303]: Can you confirm if this does not get copied to `FCS/cypress/fixtures/external` when the setup is being done?

**Review by parag-pv — APPROVED** [review]

**Review by ksentak — DISMISSED** [review]

**Review by parag-pv — DISMISSED** [review]

**Review by jmeier204 — APPROVED** [review]

**Review by ksentak — APPROVED** [review]

**Pull request merged**